packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -242,10 +265,6 @@ abstract contract PermissionManager is Initializable {
             if (_permissionId == ROOT_PERMISSION_ID || isRestricted) {
```

Comment:
Not strictly on this PR, but an optimization should be done. 

We compute isRestricted unconditionally but then check against ROOT_PERMISSION as the first contidion. 
We should check for ROOT_CONDITION as a first revert cause, and only else, compute isRestricted and check it

Reply:
Changed in 5ed4cca2287bf629a5f2e7002e8364fa295cde8e